packages/optimism-decoder/src/decoder.ts

```diff
@@ -0,0 +1,261 @@
+      result[tailA.length] = x;
+      result.set(tailB, tailA.length + 1);
+      result[tailA.length + 1 + tailB.length] = y;
+      result.set(tailC, tailA.length + 1 + tailB.length + 1);
+      result[tailA.length + 1 + tailB.length + 1 + tailC.length] = z;
+      result.set(tailD, tailA.length + 1 + tailB.length + 1 + tailC.length + 1);
+
+      if (result.length !== 4 * 31 + 3)
+        throw new Error("Assertion failed: length of result is incorrect");
+
+      const newBlobData = new Uint8Array(blobData.length + result.length);
+      newBlobData.set(blobData, 0);
+      newBlobData.set(result, blobData.length);
+      blobData = newBlobData;
+    }
+
+    datas.push(blobData.slice(4, declaredLength + 4));
+  }
+
+  const fullChannel = extractFullChannel(datas);
+
+  const decompressed = await decompressBatches(fullChannel);
+  const dataToDecode: Uint8Array = decompressed;
+  const { data: decoded } = rlp.decode(dataToDecode, true);
+
+  if (decoded[0] !== 1) {
+    throw new Error("decoded value is not a span batch");
+  }
+
+  if (!(decoded instanceof Uint8Array)) {
+    throw new Error("decode must be Uint8Array");
+  }
+
+  let currentOffset = 1;
+
+  const timestampResult = readVarint(decoded, currentOffset);
+  currentOffset = timestampResult.newOffset;
+
+  const l1OriginNumberResult = readVarint(decoded, currentOffset);
+  currentOffset = l1OriginNumberResult.newOffset;
+
+  const parentL2BlockHashResult = readBytesAsHex(decoded, currentOffset, 20);
+  currentOffset = parentL2BlockHashResult.newOffset;
+
+  const l1OriginBlockHashResult = readBytesAsHex(decoded, currentOffset, 20);
+  currentOffset = l1OriginBlockHashResult.newOffset;
+
+  const l2BlocksNumberResult = readVarint(decoded, currentOffset);
+  const l2BlocksNumber = l2BlocksNumberResult.value;
+  currentOffset = l2BlocksNumberResult.newOffset;
+
+  const originChangesResult = readBitlist(
+    l2BlocksNumber,
+    decoded,
+    currentOffset
+  );
+  const originChangesCount = originChangesResult.bits.filter(
+    (bit) => bit
+  ).length;
+  currentOffset = originChangesResult.newOffset;
+
+  let totalTxs = 0;
+  for (let i = 0; i < l2BlocksNumber; i++) {
+    const txCountResult = readVarint(decoded, currentOffset);
+    totalTxs += txCountResult.value;
+    currentOffset = txCountResult.newOffset;
+  }
+
+  const contractCreationResult = readBitlist(totalTxs, decoded, currentOffset);
+  const contractCreationTxsNumber = contractCreationResult.bits.filter(
+    (bit) => bit
+  ).length;
+  currentOffset = contractCreationResult.newOffset;
+
+  return {
+    timestampSinceL2Genesis: timestampResult.value,
+    lastL1OriginNumber: l1OriginNumberResult.value,
+    parentL2BlockHash: parentL2BlockHashResult.hex,
+    l1OriginBlockHash: l1OriginBlockHashResult.hex,
+    numberOfL2Blocks: l2BlocksNumber,
+    changedByL1Origin: originChangesCount,
+    totalTxs: totalTxs,
+    contractCreationTxsNumber: contractCreationTxsNumber,
+  } as OptimismDecodedData;
+
+  /*
```

Comment:
Debug code? should we delete it?

Reply:
I prefer to keep it. This is the implementation to decode more fields. However it is not working and I decided to leave it commented. I've added some tasks to https://github.com/Blobscan/blobscan/issues/255 to keep track of it